### PR TITLE
support  index without fields

### PR DESCRIPTION
**Pull request by panda-sheep**
As title

### CODE REVIEW

File: src/parser/test/ParserTest.cpp [hunk 1]
```diff
@@ -474,6 +474,12 @@ TEST(Parser, ColumnSpacesTest) {
 }
 
 TEST(Parser, IndexOperation) {
+    {
+        GQLParser parser;
+        std::string query = "CREATE TAG INDEX empty_field_index ON person()";
```
Comment (Shylock-Hg) [inline src/parser/test/ParserTest.cpp:479]: What's the meaning for empty field index?

Comment (panda-sheep) [reply src/parser/test/ParserTest.cpp:479]: refer to https://github.com/vesoft-inc/nebula-storage/pull/174

use to get all vertices on tag or edges on one edgetype

Comment (Shylock-Hg) [reply src/parser/test/ParserTest.cpp:479]: Emm.Mix the sentence to create properties index and tag/edge index is a little ambiguous

File: src/parser/parser.yy [hunk 2]
```diff
@@ -1736,9 +1739,11 @@ index_field
 index_field_list
     : index_field {
         $$ = new IndexFieldList();
-        std::unique_ptr<meta::cpp2::IndexFieldDef> field;
-        field.reset($1);
-        $$->addField(std::move(field));
+        if ($1) {
```
Comment (Shylock-Hg) [inline src/parser/parser.yy:1742]: I think you could remove the if statement refer to the rule like `opt_argument_list`

File: src/parser/test/ParserTest.cpp (hunk 1 above)
Comment (panda-sheep) [reply src/parser/test/ParserTest.cpp:479]: I think it's ok.
You think it just creating an index of 0 fields to get the vertices of the tag or the edges on the edgeType.

Similar to the logic of a normal index, when an index with one or more fields is included, it is not allowed to create an index with zero fields.

File: src/parser/parser.yy (hunk 2 above)
Comment (panda-sheep) [reply src/parser/parser.yy:1742]: Good, fixed!

**Review by yixinglu — COMMENTED** [review]
This PR is good to me. But I only want to know which field the index created on when user don't specify the filed of tag or edge.

**Comment by panda-sheep** [comment]
> This PR is good to me. But I only want to know which field the index created on when user don't specify the filed of tag or edge.

Good question.
We could think of it as a special index, which is only used to get the ` `of the tag or the `src`, `rank`, `dst `of the edge

**Review by yixinglu — APPROVED** [review]

**Review by laura-ding — APPROVED** [review]

**Pull request merged**